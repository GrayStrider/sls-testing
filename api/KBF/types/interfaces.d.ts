import {SubTask, Task, TaskNumber} from '../../../types/kanbanflow'

export type postReply = { taskId: string, taskNumber: TaskNumber }
export type createTaskParams =
	Omit<Task, '_id' | 'totalSecondsSpent' | 'color' | 'description'>
	& Partial<Pick<Task, 'color' | 'description'>>

interface KanbanPost {
	apiKey?: string;
}

interface ModifyTask {
	taskId: Task['_id'];
	
}

export interface CreateParams extends KanbanPost {
	params: createTaskParams;
}

export interface UpdateParams extends KanbanPost, ModifyTask {
	params: Partial<createTaskParams>;
}

type AddOrModifyPropTypes = 'subtask'
export const AddParams: { [key in AddOrModifyPropTypes]: string } = {
	subtask: 'subtasks'
}

export interface AddSubtaskParams extends KanbanPost, ModifyTask {
	params: Pick<SubTask, 'name'> & Partial<SubTask>;
	addParam: 'subtask'
}

export interface ModifySubtaskParams extends KanbanPost, ModifyTask {
	params: Partial<SubTask>;
	modifyParam: AddOrModifyPropTypes
}

export interface ImplementationParams extends KanbanPost {
	params: createTaskParams | Partial<createTaskParams>;
	taskId?: Task['_id'];
	addParam?: string
	modifyParam?: AddOrModifyPropTypes
}
