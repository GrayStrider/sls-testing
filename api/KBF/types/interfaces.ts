import {SubTask, Task, TaskNumber} from '../../../types/kanbanflow'

export type postReply = { taskId: string, taskNumber: TaskNumber }
export type CreateTaskParams =
	Omit<Task, '_id' | 'totalSecondsSpent' | 'color' | 'description'>
	& Partial<Pick<Task, 'color' | 'description'>>

interface KanbanPost {
	apiKey?: string;
}

interface ModifyTask {
	taskId: Task['_id'];
	
}

export type CreateParams2 = KanbanPost & CreateTaskParams

export interface CreateParams extends KanbanPost {
	params: CreateTaskParams;
}

export interface UpdateParams extends KanbanPost, ModifyTask {
	params: Partial<CreateTaskParams>;
}

type AddOrModifyPropTypes = 'subtask'
export const AddParams: { [key in AddOrModifyPropTypes]: string } = {
	subtask: 'subTasks'
}

export interface AddSubtaskParams extends KanbanPost, ModifyTask {
	params: Pick<SubTask, 'name'> & Partial<SubTask>;
	addParam: 'subtask'
}

export interface ModifySubtaskParams extends KanbanPost, ModifyTask {
	params: Partial<SubTask>;
	modifyParam: AddOrModifyPropTypes
}

export interface ImplementationParams extends KanbanPost {
	params: CreateTaskParams | Partial<CreateTaskParams>;
	taskId?: Task['_id'];
	addParam?: string
	modifyParam?: AddOrModifyPropTypes
}
