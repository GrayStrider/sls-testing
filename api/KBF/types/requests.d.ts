import {RequireOnlyOne} from '../../../types/types'

export type getTasksByColumnParams = byColumn & dateGrouped
export type getTasksByColumnAndSwimlaneParams =
	(byColumn & bySwimlane) & dateGrouped

interface IByColumn {
	columnId: string
	columnName: string
	columnIndex: number
}

interface IBySwimlane {
	swimlaneId: string
	swimlaneName: string
	swimlaneIndex: number
}

type byColumn = RequireOnlyOne<IByColumn,
	'columnId' | 'columnName' | 'columnIndex'>
type bySwimlane = RequireOnlyOne<IBySwimlane,
	'swimlaneId' | 'swimlaneName' | 'swimlaneIndex'>

interface dateGrouped {
	// Optional. Can only be used if column is date grouped. Task ID to start from in date grouped columns. Can not be combined with startGroupingDate.
	startTaskId?: string
	// Optional. Can only be used if column is date grouped. Valid format is YYYY-MM-DD, e.g. 2010-12-31. Can not be combined with startTaskId.
	startGroupingDate?: string
	// // Optional. Can only be used if column is date grouped. The maximum number of tasks to retrieve. The highest value that can be used is 100. Default: 20.
	limit?: number
	// Optional. Can only be used if column is date grouped. Determines in what order tasks should be searched for. Can be either asc (ascending) or desc (descending). Default: desc. Example: If using startGroupingDate=2010-12-31 and order=desc, it will get tasks for 2010-12-31 and earlier dates.
	order?: string
}
