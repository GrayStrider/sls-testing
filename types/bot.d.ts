import {Message} from 'discord.js'
import {KBFParamsGeneric} from './kanbanflow'

type ValidCommands = 'fetchTasks'
export type Commands = { [key in ValidCommands]: Command; }
export type Command = (message: Message, params?: KBFParamsGeneric) => void
