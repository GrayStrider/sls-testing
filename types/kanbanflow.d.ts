import {Attachment, Message} from 'discord.js'

export type Tasks = TasksBySwimlane[]

export interface TasksBySwimlane {
	columnId: string
	columnName: string
	tasksLimited: boolean
	tasks: Task[]
	swimlaneId: string
	swimlaneName: string
	
}

interface TaskNumber {
	value: number
	prefix?: string
}

interface SubTask {
	name: string
	finished: boolean
	dueDateTimestamp?: string
	dueDateTimestampLocal?: string
	userId?: string
}

interface Label {
	name: string
	pinned: boolean
}

interface Collaborator {
	userId: string
}


export interface RequestProps {
	comments: Comment[]
	labels: Label[]
	dates: Date[]
	collaborators: Collaborator[]
	attachments: Attachment[]
	subtasks: SubTask[]
}

interface Comment {
	_id: string
	text: string
	authorUserId: string
	createdTimestamp: string
}

export interface Task {
	// The ID of the task.
	_id: string
	// The name of the task.
	name: string
	// The description of the task.
	description: string
	// The color of the task.
	color: ColorValue
	// The ID of the column the task is in.
	columnId: string
	// The total number of seconds spent on the task. Examples: 1 hour = 3600, 2 hours = 7200 etc.
	totalSecondsSpent: number
	// The total number of seconds estimated for the task. Examples: 1 hour = 3600, 2 hours = 7200 etc. Only included if estimation unit is set to Time.
	totalSecondsEstimate?: number
	// Estimated points for the task. Only included if estimation unit is set to Points.
	pointsEstimate?: number
	// The ID of the swimlane the task is in. Only included if the board has swimlanes.
	swimlaneId?: string
	// The index of the task in the column/swimlane or date group. Only included if includePosition is set. Note: The index is zero-based.
	position?: number | 'top' | 'bottom'
	// The task number consists of an integer value and an optional prefix.
	number?: TaskNumber
	// The responsible of the task.
	responsibleUserId?: string
	// The dates of the task. Only included if there are any items.
	dates?: Date[]
	// Only included if the column the task is in is date grouped. Format is YYYY-MM-DD. Example: 2015-12-31.
	groupingDate?: string | null | ''
	// The subtasks of the task. Only included if there are any items.
	subTasks?: SubTask[]
	// The labels of the task. Only included if there are any items
	labels?: Label[]
	// The collaborators of the task. Only included if there are any items.
	collaborators?: Collaborator[]
	
	
}

export interface OtherTaskProperties {
	comments: Comment[]
	attachments: Attachment[]
	'time-entries': TimeEntry[]
	relations: Relation[]
	
}


interface Attachment {
	_id: string
	provider: string,
	name: string,
	size: number,
	mimeType: string,
	link: string,
	linkExpiresTimestamp: string,
	createdTimestamp: string,
	createdByFullName: string
}

interface TimeEntry {

}

interface Relation {

}
type Status = 'active'

type DateType = 'dueDate'

interface Date {
	targetColumnId: string
	status: Status
	dateType: DateType
	dueTimestamp?: string
	dueTimestampLocal?: string

}
type ValidCommands = 'fetchTasks'

export type Commands = { [key in ValidCommands]: Command; }
export type KBFParamsGeneric = { [key: string]: string }



export type Command = (message: Message, params?: KBFParamsGeneric) => void

interface Column {
	name: string
	uniqueId: string
}

interface Color {
	description?: string;
	name: string
	value: ColorValue
}

type ColorValue = 'yellow' | 'white' | 'red' | 'green' | 'blue' | 'purple' | 'orange' | 'cyan' | 'brown' | 'magenta'

interface Swimlane {
	name: string
	uniqueId: string
}
export interface Board {
	name: string
	columns: Column[]
	_id: string
	colors: Color[]
	swimlanes: Swimlane[]
}
