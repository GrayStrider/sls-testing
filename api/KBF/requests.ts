import {Board, KBFParamsGeneric, Task, Tasks} from '../../types/kanbanflow'
import {kanbanGet} from './KBF'

const getBoard = () => kanbanGet<Board>({resource: 'board'})
const getTaskByID = (id: Task['_id']) => kanbanGet<Task>({resource: `tasks/${id}`})
const getTasksByColumn = (params: getTasksByColumnParams) =>
	kanbanGet<Tasks>({resource: 'tasks', params: params})



export interface getTasksByColumnParams {
	columnId: string
}

export {getBoard, getTaskByID, getTasksByColumn}
