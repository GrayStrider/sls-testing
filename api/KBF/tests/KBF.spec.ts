import {Attachment, Board, Comment, Task, Tasks, TasksBySwimlane} from '../../../types/kanbanflow'
import {createTask, getAllTasksFromBoard, getBoard, getTaskByID, getTaskDetailsById, getTasksByColumn, getTasksByColumnAndSwimlane, postParams} from '../requests'
import {maxFeatuesId, taskMaxFeatures, taskMinFeatues, testColumnId, testDate, testLabel, testSubtasks, testSwimlaneId, testUserId} from './mocks'

it('should fetch valid board data', async () => {
	const exp: Board = {
		_id: 'is54ys',
		name: 'GH> KBF TS',
		columns: [
			{
				uniqueId: 'UpPKrbzD8yBN',
				name: 'To-do'
			},
			{
				uniqueId: 'Uqsc6jy2Cbl9',
				name: 'TEST'
			},
			{
				uniqueId: 'Urnu6XfQg3yU',
				name: 'Awaiting review'
			},
			{
				uniqueId: 'UsbzO1HUaPhw',
				name: 'Done'
			}
		],
		colors: [
			{
				name: 'White',
				description: 'desc',
				value: 'white'
			}
		],
		swimlanes: [
			{
				uniqueId: 'V8pP3mn7NcSG',
				name: 'A'
			},
			{
				uniqueId: 'V9eKUkwDY8Vz',
				name: 'B'
			}
		]
	}
	const res = await getBoard()
	expect(res).toMatchObject(exp)
	
})
it('should return a single task by id', async () => {
	
	const act1: Task = await getTaskByID(maxFeatuesId)
	expect(act1).toMatchObject(taskMaxFeatures)
	const act2: Task = await getTaskByID('hh6RHiEw')
	expect(act2).toMatchObject(taskMinFeatues)
})
it('should return all tasks in the column', async () => {
	const expTasks: Tasks = [
		{
			columnId: 'Uqsc6jy2Cbl9',
			columnName: 'TEST',
			tasksLimited: false,
			tasks: [taskMinFeatues, taskMaxFeatures],
			swimlaneId: 'V8pP3mn7NcSG',
			swimlaneName: 'A',
		},
		{
			columnId: 'Uqsc6jy2Cbl9',
			columnName: 'TEST',
			tasksLimited: false,
			tasks: [],
			swimlaneId: 'V9eKUkwDY8Vz',
			swimlaneName: 'B'
		}
	]
	const exp = expect.arrayContaining([
		expect.objectContaining(expTasks[0]),
		expect.objectContaining(expTasks[1])
	])
	
	const actById: Tasks =
		await getTasksByColumn({columnId: 'Uqsc6jy2Cbl9'})
	expect(actById[0]).toMatchObject(expTasks[0])
	expect(actById[1]).toMatchObject(expTasks[1])
	
	const actByName: Tasks =
		await getTasksByColumn({columnName: 'TEST'})
	expect(actByName).toEqual(exp)
	
	const actByIndex: Tasks =
		await getTasksByColumn({columnIndex: 1})
	expect(actByIndex).toEqual(exp)
	
	const expTaskByColumnAndSwimlane: TasksBySwimlane = {
		columnId: 'Uqsc6jy2Cbl9',
		columnName: 'TEST',
		tasksLimited: false,
		tasks: [],
		swimlaneId: 'V9eKUkwDY8Vz',
		swimlaneName: 'B'
	}
	const expByColumnAndSwimlane = expect.arrayContaining([
		expect.objectContaining(expTaskByColumnAndSwimlane)
	])
	const actByColumnAndSwimlane =
		await getTasksByColumnAndSwimlane({columnIndex: 1, swimlaneId: 'V9eKUkwDY8Vz'})
	expect(actByColumnAndSwimlane).toMatchObject(expByColumnAndSwimlane)
	
})
it('should return all tasks on the board', async () => {
	const res = await getAllTasksFromBoard()
	// snapshot will do, since all types are tested in previous tests.
	expect(res).toMatchSnapshot()
})
describe('should fetch specific task properties', () => {
	it('should fetch comments', async () => {
		const act = await getTaskDetailsById(maxFeatuesId, 'comments')
		
		const exp: Comment = {
			_id: 'm8q9JVny',
			text: 'test',
			authorUserId: '5b2a3ad796a22ce0d687f7f8181232b1',
			createdTimestamp: '2019-11-01T04:21:31.774Z'
		}
		
		expect(act).toMatchObject(
			expect.arrayContaining([
				expect.objectContaining(exp)
			])
		)
		
	})
	it('should fetch labels', async () => {
		const act = await getTaskDetailsById(maxFeatuesId, 'labels')
		
		expect(act).toMatchObject(
			expect.arrayContaining([
				expect.objectContaining(testLabel)
			])
		)
	})
	it('should fetch dates', async () => {
		const act = await getTaskDetailsById(maxFeatuesId, 'dates')
		expect(act).toMatchObject(
			expect.arrayContaining([
				expect.objectContaining(testDate)
			
			])
		)
	})
	it('should fetch subtasks', async () => {
		const act = await getTaskDetailsById(maxFeatuesId, 'subtasks')
		expect(act[0]).toMatchObject(testSubtasks[0])
		expect(act[1]).toMatchObject(testSubtasks[1])
	})
	it('should fetch collaborators', async () => {
		const act = await getTaskDetailsById(maxFeatuesId, 'collaborators')
		
		expect(act).toHaveLength(0)
		
	})
	it('should fetch attachments', async () => {
		const act = await getTaskDetailsById(maxFeatuesId, 'attachments')
		const exp: Partial<Attachment> = {
			// some properties omitted
			_id: 'QWAnJzYkcu7x',
			provider: 'KanbanFlow',
			name: 'wuauserv.reg.jpg',
			size: 6990,
			mimeType: 'image/jpeg',
			createdTimestamp: '2019-11-01T12:37:01.675Z',
		}
		expect(act[0]).toMatchObject(exp)
	})
})


describe('should create tasks/properties', () => {
	it('should add new task', async () => {
		const minFeaturesParams: postParams = {
			name: 'TESTMIN',
			columnId: testColumnId,
			swimlaneId: testSwimlaneId, //todo warning on ommiting swimlane when board has swimlanes, or implement check before submitting task!
		}
		const maxFeaturesParams: postParams = {
			name: 'TESTMAX',
			columnId: testColumnId,
			swimlaneId: testSwimlaneId, //todo warning on ommiting swimlane when board has swimlanes, or implement check before submitting task!
			// position: 'bottom', //todo position and grouping data are mutually exclusive
			color: 'green',
			description: 'TEST',
			number: {prefix: 'TESTPREFIX', value: 99},
			responsibleUserId: testUserId,
			totalSecondsEstimate: 60,
			pointsEstimate: 100.99,
			groupingDate: null,
			dates: [testDate],
			subTasks: [testSubtasks[0]],
			labels: [testLabel],
			collaborators: []
		}
		
		const actMin = await createTask(minFeaturesParams)
		expect(actMin).toHaveProperty(['taskId'])
		expect(actMin).toHaveProperty(['taskNumber'])
		
		const actMax = await createTask(maxFeaturesParams)
		expect(actMax).toHaveProperty(['taskId'])
		expect(actMax).toHaveProperty(['taskNumber'])
	})
	it('should create subtask', async () => {
		
		
		throw 'not implemented'
	})
	it('should create label', async () => {
		throw 'not implemented'
	})
	it('should create / update date', async () => {
		throw 'not implemented'
	})
	it('should create add collaborator', async () => {
		throw 'not implemented'
	})
	it('should create comment', async () => {
		throw 'not implemented'
	})
	it('should add attachment', async () => {
		throw 'not implemented'
	})
	it('should add manual time entry', async () => {
		throw 'not implemented'
	})
	
})

describe('should update entries', () => {
	it('should update subtasks', async () => {
		throw 'not implemented'
	})
	it('should update label', async () => {
		throw 'not implemented'
	})
	it('should create / update date', async () => {
		throw 'not implemented'
	})
	it('update comment', async () => {
		throw 'not implemented'
	})
	it('should update manual time entry', async () => {
		throw 'not implemented'
	})
	
})

describe('should delete entries', () => {
	it('should delete subtask', async () => {
		
		
		throw 'not implemented'
	})
	it('should delete label', async () => {
		throw 'not implemented'
	})
	it('should delete date', async () => {
		throw 'not implemented'
	})
	it('should delete collaborator', async () => {
		throw 'not implemented'
	})
	it('should delete comment', async () => {
		throw 'not implemented'
	})
	it('should delete attachment', async () => {
		throw 'not implemented'
	})
	it('should delete manual time entry', async () => {
		throw 'not implemented'
	})
	
})

describe('should manage time entries', () => {
	it('should get stoppwatch entries', async () => {
		
		
		throw 'not implemented'
	})
	it('should delete stopwatch entries', async () => {
		throw 'not implemented'
	})
	it('should get pomodory entries', async () => {
		throw 'not implemented'
	})
	it('should delete pomodoro entries', async () => {
		throw 'not implemented'
	})
	it('should get manual entries', async () => {
		throw 'not implemented'
	})
	it('should delete manual entries', async () => {
		throw 'not implemented'
	})
	
})

it('should get users', async () => {
  
  throw 'Not implemented'
});
it('should get events', async () => {
  
  throw 'Not implemented'
});

describe('should manage webhooks', () => {
	it('should create webhook', async () => {
	  
	  throw 'Not implemented'
	});
	it('should update webhook', async () => {
		
		throw 'Not implemented'
	});
	it('should delete webhoook', async () => {
		
		throw 'Not implemented'
	});
	it('should recieve events', async () => {
		
		throw 'Not implemented'
	});
	it('should verify signature', async () => {
		
		throw 'Not implemented'
	});
	
})
