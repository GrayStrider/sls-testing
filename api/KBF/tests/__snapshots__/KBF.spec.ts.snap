// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fetch data comments 1`] = `
Array [
  Object {
    "columnId": "UpPKrbzD8yBN",
    "columnName": "To-do",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [
      Object {
        "_id": "hknJb3Jy",
        "color": "white",
        "columnId": "UpPKrbzD8yBN",
        "description": "",
        "name": "swim2",
        "number": Object {
          "prefix": "PRE-",
          "value": 2,
        },
        "swimlaneId": "V8pP3mn7NcSG",
        "totalSecondsEstimate": 0,
        "totalSecondsSpent": 0,
      },
    ],
    "tasksLimited": false,
  },
  Object {
    "columnId": "UpPKrbzD8yBN",
    "columnName": "To-do",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [
      Object {
        "_id": "hkoDXmD6",
        "color": "white",
        "columnId": "UpPKrbzD8yBN",
        "description": "",
        "name": "swim1",
        "number": Object {
          "prefix": "PRE-",
          "value": 3,
        },
        "swimlaneId": "V9eKUkwDY8Vz",
        "totalSecondsEstimate": 0,
        "totalSecondsSpent": 0,
      },
    ],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Uqsc6jy2Cbl9",
    "columnName": "TEST",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [
      Object {
        "_id": "hh6RHiEw",
        "color": "white",
        "columnId": "Uqsc6jy2Cbl9",
        "description": "",
        "groupingDate": "",
        "name": "test_task",
        "swimlaneId": "V8pP3mn7NcSG",
        "totalSecondsEstimate": 0,
        "totalSecondsSpent": 0,
      },
      Object {
        "_id": "hia9zFNn",
        "color": "white",
        "columnId": "Uqsc6jy2Cbl9",
        "dates": Array [
          Object {
            "dateType": "dueDate",
            "dueTimestamp": "2019-11-01T06:00:00.000Z",
            "dueTimestampLocal": "2019-11-01T17:00:00+11:00",
            "status": "active",
            "targetColumnId": "UsbzO1HUaPhw",
          },
        ],
        "description": "test",
        "groupingDate": "",
        "labels": Array [
          Object {
            "name": "test_label",
            "pinned": false,
          },
        ],
        "name": "test_task3",
        "number": Object {
          "prefix": "PRE-",
          "value": 1,
        },
        "responsibleUserId": "5b2a3ad796a22ce0d687f7f8181232b1",
        "subTasks": Array [
          Object {
            "dueDateTimestamp": "2019-11-01T06:00:00.000Z",
            "dueDateTimestampLocal": "2019-11-01T17:00:00+11:00",
            "finished": false,
            "name": "sub1",
          },
          Object {
            "finished": false,
            "name": "sub2",
            "userId": "5b2a3ad796a22ce0d687f7f8181232b1",
          },
        ],
        "swimlaneId": "V8pP3mn7NcSG",
        "totalSecondsEstimate": 60,
        "totalSecondsSpent": 60,
      },
    ],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Uqsc6jy2Cbl9",
    "columnName": "TEST",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Urnu6XfQg3yU",
    "columnName": "Awaiting review",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Urnu6XfQg3yU",
    "columnName": "Awaiting review",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "UsbzO1HUaPhw",
    "columnName": "Done",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "UsbzO1HUaPhw",
    "columnName": "Done",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [],
    "tasksLimited": false,
  },
]
`;

exports[`should fetch data labels 1`] = `
Array [
  Object {
    "columnId": "UpPKrbzD8yBN",
    "columnName": "To-do",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [
      Object {
        "_id": "hknJb3Jy",
        "color": "white",
        "columnId": "UpPKrbzD8yBN",
        "description": "",
        "name": "swim2",
        "number": Object {
          "prefix": "PRE-",
          "value": 2,
        },
        "swimlaneId": "V8pP3mn7NcSG",
        "totalSecondsEstimate": 0,
        "totalSecondsSpent": 0,
      },
    ],
    "tasksLimited": false,
  },
  Object {
    "columnId": "UpPKrbzD8yBN",
    "columnName": "To-do",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [
      Object {
        "_id": "hkoDXmD6",
        "color": "white",
        "columnId": "UpPKrbzD8yBN",
        "description": "",
        "name": "swim1",
        "number": Object {
          "prefix": "PRE-",
          "value": 3,
        },
        "swimlaneId": "V9eKUkwDY8Vz",
        "totalSecondsEstimate": 0,
        "totalSecondsSpent": 0,
      },
    ],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Uqsc6jy2Cbl9",
    "columnName": "TEST",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [
      Object {
        "_id": "hh6RHiEw",
        "color": "white",
        "columnId": "Uqsc6jy2Cbl9",
        "description": "",
        "groupingDate": "",
        "name": "test_task",
        "swimlaneId": "V8pP3mn7NcSG",
        "totalSecondsEstimate": 0,
        "totalSecondsSpent": 0,
      },
      Object {
        "_id": "hia9zFNn",
        "color": "white",
        "columnId": "Uqsc6jy2Cbl9",
        "dates": Array [
          Object {
            "dateType": "dueDate",
            "dueTimestamp": "2019-11-01T06:00:00.000Z",
            "dueTimestampLocal": "2019-11-01T17:00:00+11:00",
            "status": "active",
            "targetColumnId": "UsbzO1HUaPhw",
          },
        ],
        "description": "test",
        "groupingDate": "",
        "labels": Array [
          Object {
            "name": "test_label",
            "pinned": false,
          },
        ],
        "name": "test_task3",
        "number": Object {
          "prefix": "PRE-",
          "value": 1,
        },
        "responsibleUserId": "5b2a3ad796a22ce0d687f7f8181232b1",
        "subTasks": Array [
          Object {
            "dueDateTimestamp": "2019-11-01T06:00:00.000Z",
            "dueDateTimestampLocal": "2019-11-01T17:00:00+11:00",
            "finished": false,
            "name": "sub1",
          },
          Object {
            "finished": false,
            "name": "sub2",
            "userId": "5b2a3ad796a22ce0d687f7f8181232b1",
          },
        ],
        "swimlaneId": "V8pP3mn7NcSG",
        "totalSecondsEstimate": 60,
        "totalSecondsSpent": 60,
      },
    ],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Uqsc6jy2Cbl9",
    "columnName": "TEST",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Urnu6XfQg3yU",
    "columnName": "Awaiting review",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "Urnu6XfQg3yU",
    "columnName": "Awaiting review",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "UsbzO1HUaPhw",
    "columnName": "Done",
    "swimlaneId": "V8pP3mn7NcSG",
    "swimlaneName": "A",
    "tasks": Array [],
    "tasksLimited": false,
  },
  Object {
    "columnId": "UsbzO1HUaPhw",
    "columnName": "Done",
    "swimlaneId": "V9eKUkwDY8Vz",
    "swimlaneName": "B",
    "tasks": Array [],
    "tasksLimited": false,
  },
]
`;
